### PR TITLE
add decorator to require login in several places

### PR DESCRIPTION
**Pull request by presnick**

### CODE REVIEW

File: controllers/assignments.py [hunk 1]
```diff
@@ -487,7 +487,7 @@ def get_source(acid):
         res['file_includes'] = [{'acid': acid, 'contents': get_source(acid)} for acid in file_divs]
     return json.dumps(res)
 
-
+@auth.requires_login()
```
Comment (bnmnetp) [inline controllers/assignments.py]: The decorator is redundant with the if not auth.user check right below.

Comment (presnick) [reply controllers/assignments.py]: Yes, but there are a couple functions that didn't have it, where I added it. And I think we should probably use the decorator everywhere and get rid of the check inside the code.

Comment (bjones1) [reply controllers/assignments.py]: I agree -- this is good DRY coding (use a single-line decorator, instead of multi-line checks).

Comment (presnick) [reply controllers/assignments.py]: I have removed the redundant auth checks now.

**Pull request merged**